### ISSUE TITLE
HDF5 error and APS1 h5 file loading error?

### ISSUE THREAD

**Issue by jaseung** [BBN-Q/Auspex#248]
I am trying to set up Auspex to work with APS1 and AlazarATS9870. While I am running a simple cavity scan, I noticed HDF5 errors in cmd console. Here are my Auspex script and screenshot showing HDF errors. 

![auspex_script](https://user-images.githubusercontent.com/22563147/39636504-6893b900-4f8e-11e8-8edb-3517d20f4007.PNG)
![hdf5_error](https://user-images.githubusercontent.com/22563147/39636665-ee854de4-4f8e-11e8-9ea5-6b480a02acc8.png)

Also, I guess it might be related with the first issue, but it seems that APS1 is not outputting anything. I ran unit_test_basic function from aps.py and it failed in loading a sequence file. Below are the screenshot of jupyter notebook and cmd console output.

![unit_test_aps1_2](https://user-images.githubusercontent.com/22563147/39645333-fa82d3ca-4fa5-11e8-9380-c8afa59ff72b.PNG)

![unit_test_aps1_cmd](https://user-images.githubusercontent.com/22563147/39637105-0fc56934-4f90-11e8-8412-c288b6db63c5.PNG)

Those errors point to C:/git/mingw/mingw-w64-hdf5/src/hdf5-1.8.14/src folder, but I don't have it in my computer. This might be the problem? Currently, I am using libaps fix/py3-compat branch and libaps.dll from libaps release ver 1.3.2.  Qlab works fine with the same dll.

**Comment by jaseung** [comment]
I tried building the driver from the source of libaps fix/py3-compat using MSYS2 and MinGW-W64, and the result is basically the same; the same errors. I can't load h5 file into APS1. I have no idea what to look further.

**Comment by dieris** [comment]
Have you tried to load it directly using the `libaps` instructions, instead of Auspex? I believe that @gribeill and I had the same issue https://github.com/BBN-Q/Auspex/pull/203

**Comment by jaseung** [comment]
Yes, I have, but still there was h5 loading error. However, I will try
again just in case I might have made some changes.

Thanks,
Jaseung

**Comment by jaseung** [comment]
No luck. You can see the screenshot for loading h5 using the libaps
methods. There seems no error in library call for 'load_sequence_file' in
jupyter-notebook, but hdf5 errors are caught and shown in the cmd console.
Was anyone at BBN able to run a sequence file in APS1 using libaps python
recently, i.e. pulse is created?

Jaseung

**Comment by gribeill** [comment]
Do you also have libaps2 installed?

**Comment by gribeill** [comment]
As a further update, I am able to play back sequence files but with a different version of HDF5. I think this is a windows DLL mess, but it's not clear to me why this particular combination of HDF5 libraries is not working.

**Comment by gribeill** [comment]
One more question for you @jaseung: can you please tell me the output of `import h5py; h5py.version.hdf5_version` in your python?

**Comment by jaseung** [comment]
1. Yes, I do have libaps2 installed.
2. '1.10.1' =  output of import h5py; h5py.version.hdf5_version

If this is a hdf dll issue, I imagine building the driver using the same hdf5 version as the one in our current Matlab should work? libaps.dll still works in Qlab, i.e. can load h5 files.

**Comment by gribeill** [comment]
It's not clear to me which DLL is causing the problem, so I am still investigating. In the meantime, can you please let me know the version and path information for HDF5 returned by CMake (using `cmake -G "MSYS Makefiles" /path/to/src`?

**Comment by jaseung** [comment]
I am not sure if this is what you want. 
![hdf5_version](https://user-images.githubusercontent.com/22563147/39818775-ad259072-536f-11e8-82fb-4cff3c7cdcbf.PNG)

To be on the same page.. In both cases where I used libaps-v1.3.2 release version and I built things from the source from libaps fix/py3-compat branch, I saw the same hdf errors. The hdf version used to build the release seems to be 1.8.14 (based on the error message), and the hdf version in MSYS2 MinGW64 to be 1.8.20.

**Comment by gribeill** [comment]
I'll be traveling till Monday so won't have much time to debug this till next week. Some thoughts:

- run `ldd libaps.dll` in MSYS2 to check that all DLLs are found and that they are being loaded from where you think they should be
- you could try downgrading your anaconda hdf5 to 1.8 and see if the HDF5 files will be loaded.

**Comment by jaseung** [comment]
I tried both, but had the same hdf5 error.

1. DLLs seem to be placed and loaded as expected.

![ldd_libaps dll](https://user-images.githubusercontent.com/22563147/39929401-f06c9d02-5505-11e8-99ac-dbd5f048151c.png)

2. I downgraded hdf5 from 2.7.1. to 2.7.0, which has hdf5 version of 1.8.18.

I wonder if you guys also see the similar hdf5 errors when loading a h5 file to APS1 in python3 or it's just for us here.

**Comment by jaseung** [comment]
1. I think APS1 just worked. The "unit_test_basic" function in aps.py worked with a new h5 file that I created using QGL (I can see the right waveform using an oscilloscope). Somehow the UnitTest.h5 file provided in src/lib folder doesn't work and I think it should be replaced. 

2. The h5 errors that made me think APS1 can't load a h5 file seem harmless. There seemed a similar issue here although I don't quite follow what they are talking about a thread yet: 
https://github.com/h5py/h5py/issues/580 

3. When I run Auspex scrips fresh, APS1 doesn't seem to output anything. Only after I manually enable the output for example, using set_enable method in aps.py, I can see a waveform coming out of APS1. I am not certain if this is just a bug or not. The Auspex script that I ran is the following:

from QGL import 
from auspex.exp_factory import QubitExpFactory
cl = ChannelLibrary()
q1 = QubitFactory("q1")
exp = QubitExpFactory.create(PulsedSpec(q1))
exp.add_qubit_sweep("q1 measure frequency", np.linspace(6e9, 6.5e9, 4))
exp.run_sweeps()
